File: java/src/main/java/demo/officeDay/defaultMethod/FlyingCar.java
package demo.officeDay.defaultMethod;

public class FlyingCar implements CarInfo, AirplaneInfo {

    @Override
    public String getMaker() {
        return "This is to solve multiple inheritance method conflict";
    }

    @Override
    public int getMaxSpeed() {
        return 140;
    }

    @Override
    public int getMaxAltitude() {
        return 2000;
    }
}

File: csharp/demo.officeDay/defaultMethod/FlyingCar.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace demo.officeDay.defaultMethod
{
    public class FlyingCar:AirplaneInfo,CarInfo
    {
        const string UNKNOWN_MAKER = "Unknown Maker";

        const string UNKNOWN_MODEL = "Unknown Model";

        public string Model
        {
            get { return UNKNOWN_MODEL; }
        }

        public string Maker
        {
            get { return UNKNOWN_MAKER; }
        }
        public int getMaxSpeed()
        {
            throw new NotImplementedException();
        }

        public int getMaxAltitude()
        {
            throw new NotImplementedException();
        }

        /*
        string AirplaneInfo.Model
        {
            get { return UNKNOWN_MODEL; }
        }
        
        string CarInfo.Model
        {
            get { return UNKNOWN_MODEL; }
        }
        */
    }
}
